swarm/network/stream/snapshot_sync_test.go

```diff
@@ -0,0 +1,568 @@
+					err = r.RequestSubscription(conf.addrToIdMap[string(a.Address())], NewStream("SYNC", []byte{byte(uint8(c))}, false), Top)
+					if err != nil {
+						log.Error(fmt.Sprintf("Error subscribing! %v", err))
+						return false
+					}
+					remember[c] = true
+				}
+				return true
+			})
+		}
+		prev = po
+		return true
+	})
+
+	return nil
+}
+
+func checkChunkIsAtNode(conf *synctestConfig) {
+	allOk := true
+	for chunk, nodes := range conf.chunksToNodesMap {
+		for _, node := range nodes {
+			if ok, _ := stores[conf.addrToIdMap[string(conf.addrs[node])]].Get([]byte(chunk)); ok != nil {
+				if len(conf.retrievalMap[chunk]) == 0 {
+					conf.retrievalMap[chunk] = make(map[string]time.Duration)
+				}
+				conf.retrievalMap[chunk][string(conf.addrs[node])] = time.Since(startTime)
+			}
+			if conf.retrievalMap[chunk][string(conf.addrs[node])] == 0 {
+				allOk = false
+			}
+		}
+	}
+	if allOk {
+		log.Info("All chunks arrived at destination")
+		for ch, n := range conf.retrievalMap {
+			for a, t := range n {
+				log.Info(fmt.Sprintf("Chunk %s at node %s took %d ms", string(ch), string(a), t.Seconds()*1e3))
+			}
+		}
+	}
+}
+
+//map chunk keys to addresses which are responsible
+func mapKeysToNodes(conf *synctestConfig) *synctestConfig {
+	kmap := make(map[string][]int)
+	nodemap := make(map[string][]int)
+	//build a pot for chunk hashes
+	np := pot.NewPot(nil, 0)
+	mm := make(map[string]int)
+	for i, a := range conf.addrs {
+		mm[string(a)] = i
+		np, _, _ = pot.Add(np, a, pof)
+	}
+	//for each address, run EachNeighbour on the chunk hashes pot to identify closest nodes
+	fmt.Println(conf.chunks)
+	for i := 0; i < len(conf.chunks); i++ {
+		pl := 256 //highest proximity
+		var nns []int
+		np.EachNeighbour([]byte(conf.chunks[i]), pof, func(val pot.Val, po int) bool {
+			a := val.([]byte)
+			if pl == 256 || pl == po {
+				fmt.Println(fmt.Sprintf("appending %s", conf.addrToIdMap[string(a)]))
+				nns = append(nns, mm[string(a)])
+				nodemap[string(a)] = append(nodemap[string(a)], i)
+			}
+			if pl == 256 && len(nns) >= testMinProxBinSize {
+				pl = po
+			}
+			return true
+		})
+		kmap[conf.chunks[i].String()] = nns
+		//log.Debug(fmt.Sprintf("Length for id %s: %d",ids[i],len(kmap[ids[i]])))
+	}
+	for k, v := range nodemap {
+		fmt.Print(fmt.Sprintf("Node %s: ", conf.addrToIdMap[k]))
+		for _, vv := range v {
+			fmt.Println(conf.chunks[vv])
+		}
+		fmt.Println(conf.addrToIdMap[k])
+		fmt.Println("-------------------------------")
+	}
+	for k, v := range kmap {
+		fmt.Print(fmt.Sprintf("Chunk %s: ", k))
+		for _, vv := range v {
+			fmt.Println(conf.addrToIdMap[string(conf.addrs[vv])])
+		}
+		fmt.Println("###############################")
+	}
+	conf.nodesToChunksMap = nodemap
+	conf.chunksToNodesMap = kmap
+	return conf
+}
+
+//upload a file(chunks) to a single local node store
+func uploadFileToSingleNodeStore(id discover.NodeID, chunkCount int) ([]storage.Key, error) {
+	log.Debug(fmt.Sprintf("Uploading to node id: %s", id))
```

Comment:
we dont need the dpa here. why not just create a chunk and store it. 
There will be a `newChunkFromData` func

Reply:
@zelig aren't these high level tests? So maybe keeping the `dpa` isn't that bad, after all I just copy pasted from the `syncer_test.go` file. Also, wondering if we later integrate encryption, if that would affect correct functioning of these tests if we don't use `dpa`?